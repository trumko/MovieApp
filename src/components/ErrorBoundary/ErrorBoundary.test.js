import React from 'react'
import { shallow } from 'enzyme'
import ErrorBoundary from './ErrorBoundary'

describe('ErrorBoundary', () => {
  it('should render the warning when an error occurs', () => {
    const component = shallow(
      <ErrorBoundary className="testClass">
        <p>Children components</p>
      </ErrorBoundary>
    );
    component.setState({ hasError: true });
    expect(component).toMatchSnapshot();
  });

  it('should render the regular content when there is no error', () => {
    const component = shallow(
      <ErrorBoundary className="testClass">
        <p>Children components</p>
      </ErrorBoundary>
    );
    component.setState({ hasError: false });
    expect(component).toMatchSnapshot();
  });
});
