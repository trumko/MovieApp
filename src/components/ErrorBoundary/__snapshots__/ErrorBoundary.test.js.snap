// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary should render the regular content when there is no error 1`] = `
<p>
  Children components
</p>
`;

exports[`ErrorBoundary should render the warning when an error occurs 1`] = `
<h1>
  Something went wrong
</h1>
`;
